### DISCUSSION TITLE
Related products?

### DISCUSSION THREAD

**Discussion by trnc-ck** [vendurehq/vendure#2178] [General]
hi,

is there a way to add related products to a product / product variant so we can display them next to the product on a product detail page? Didn't find anything in the admin.

**Comment by WeAreHaus** [comment]
We would like to do this as well. Is it possible out of the box or is customization needed?

**Comment by michaelbromley** [comment]
Hi, this is something you can do using [custom fields](https://docs.vendure.io/developer-guide/customizing-models/):

```ts
import { Product, VendureConfig } from '@vendure/core';

const config: VendureConfig = {
  // ....
  customFields: {
    Product: [
      {
        name: 'relatedProducts',
        type: 'relation',
        entity: Product,
        list: true,
        ui: { component: 'product-multi-form-input', selectionMode: 'product' }
      }
    ],
  }
}
```

**Reply by WeAreHaus** [reply]
Cool, thanks!

**Reply by nonlinearcom** [reply]
Hi @WeAreHaus, did you manage to get it working? 

In my case (V2.0.2) after adding related products the update button remains disabled. Then, if edit any other field and try to update i get:

```js
Variable "$input" got invalid value null at "input.customFields.relatedProductsIds[0]"; Expected non-nullable type "ID!" not to be null. 
```

**Reply by oidt** [reply]
Same error here, without the UI in the json it is working for me in a other kind of view. The ui component also looks like to show deleted products I do not see in my products overview. When I have the time I will make a bug for it.

**Reply by michaelbromley** [reply]
Fixing coming soon in v2.0.3
